### PR TITLE
adapted to changed jupyterhub architecture

### PR DESCRIPTION
**Pull request by dennisbuitendijk**
This is a fix to work with jupyterhub's new permission system.

### CODE REVIEW

File: jupyterhub/handlers/pages.py [hunk 1]
```diff
@@ -455,7 +456,9 @@ class AdminHandler(BaseHandler):
     """Render the admin page."""
 
     @web.authenticated
-    @admin_only
+    @needs_scope('users')  # stacked decorators: all scopes must be present
+    @needs_scope('admin:users')
+    @needs_scope('admin:servers')
```
Comment (ArquintL) [inline jupyterhub/handlers/pages.py]: what is the difference to `@admin_only`? Is the new scope more or less permissive?

**Comment by ArquintL** [comment]
Also, this PR contains a conflict

File: jupyterhub/handlers/pages.py (hunk 1 above)
Comment (dennisbuitendijk) [reply jupyterhub/handlers/pages.py]: `@admin_only` was removed from jupyterhub and was replaced by the scope module.

**Comment by dennisbuitendijk** [comment]
I have resolved the merge conflict and fixed an issue where my additions to Code Server could not read the cookie.
@ArquintL I found an issue with using the logout button in Code Server when running with the Hub where the proxy does not find the backend service anymore. I cannot find the reason for that.

**Review by ArquintL — DISMISSED** [review]

**Comment by ArquintL** [comment]
> @ArquintL I found an issue with using the logout button in Code Server when running with the Hub where the proxy does not find the backend service anymore. I cannot find the reason for that.

Is this a regression introduced by this PR? What next steps do you suggest?
I've approved this PR now but leave it up to you to merge it

**Comment by dennisbuitendijk** [comment]
> > @ArquintL I found an issue with using the logout button in Code Server when running with the Hub where the proxy does not find the backend service anymore. I cannot find the reason for that.
> 
> Is this a regression introduced by this PR? What next steps do you suggest?
> I've approved this PR now but leave it up to you to merge it

I do not know why the old way did not work but I found the correct target for the route now.

**Review by ArquintL — APPROVED** [review]

**Pull request merged**